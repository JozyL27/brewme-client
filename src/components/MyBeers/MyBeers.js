import React from 'react'
import SearchBeerService from '../../services/search-beer-service'
import BeerContext from '../../context/beerContext'

export default class MyBeers extends React.Component {
    static contextType = BeerContext

    render() {
        return (
            <>
            <div>im beers!</div>
            </>
        )
    }
}