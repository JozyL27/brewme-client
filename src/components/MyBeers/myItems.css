.myUl {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 0px;
}

.myLi {
    width: 70%;
    text-align: center;
    margin: 15px auto 15px auto;
    padding: 15px;
    /* gradient below */
    border-top: 5px solid rgb(135,206,250);
    border-bottom: 5px solid #E1C340;
    background-image: 
        linear-gradient(rgb(135,206,250), #E1C340),
        linear-gradient(rgb(135,206,250), #E1C340)
    ;
    background-size:5px 100%;
    background-position:0 0, 100% 0;
    background-repeat:no-repeat;
    border-radius: 4px;
    font-weight: bold;
    color: white;
    line-height: 1.5;
}



.myAbv {
    margin: 10px auto 5px auto;
}


.deleteButton {
    position: relative;
    right: 5px;
    color: white;
    padding: 6px 18px;
    text-decoration: none;
    display: inline-block;
    text-align: center;
    font-size: 16px;
    margin: 40px auto 25px auto;
    transition-duration: 0.4s;
    cursor: pointer;
    background-color: rgb(135,206,250); 
    border: 2px solid #191919;
    border-radius: 8px;
    font-weight: bold;
}

.deleteButton:hover {
    background-color: red;
    color: white;
}

.noBeersPara {
    margin-top: 35px;
    font-weight: bold;
    color: red;
    text-align: center;
}

.myBeersError {
    font-weight: bold;
    color: red;
    text-align: center;
}

/* do not use width in media q */

/* tablets */
@media screen and (min-width: 768px) {
    .myLi {
        width: 60%;
    }
}


/* laptops */
@media screen and (min-width: 1024px) {
    .myLi {
        width: 40%;
    }
}