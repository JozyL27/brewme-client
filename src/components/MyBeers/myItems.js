import React from 'react'



export default function MyItems(props) {
    function checkForDescript(descript) {
        if(descript.length <= 1) {
            return  <img src="https://cdn11.bigcommerce.com/s-nq6l4syi/images/stencil/1280x1280/products/16517/21038/71882-1024__10025.1516616292.jpg?c=2?imbypass=on" alt="bookimg" />
        } 
    
        return <p>{descript}</p>
    }

    return (
        <>
        {props.brewskis.map(beer => <li key={beer.beer_id}> {beer.name}
            <div>{beer.abv}</div>
            <p>{beer.descript}</p>
        </li>)}
        </>
    )
}