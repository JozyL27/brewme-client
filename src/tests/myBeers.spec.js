import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MyBeers from '../components/MyBeers/MyBeers'


describe(`MyBeers component`, () => {
    
    it(`MyBeers component renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<MyBeers />, div)
        ReactDOM.unmountComponentAtNode(div)
    })
})