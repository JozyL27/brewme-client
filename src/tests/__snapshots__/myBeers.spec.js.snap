// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBeers component renders the UI as expected 1`] = `
Array [
  <p>
    There are currently no beers in your list!
  </p>,
  <ul
    className="myUl"
  />,
]
`;
